migrations/0029-add-order-pickupEvent-field.ts

```diff
@@ -6,12 +6,14 @@ export class AddOrderPickupEventField1631941813500 implements MigrationInterface
   public async up(queryRunner: QueryRunner): Promise<void> {
     await queryRunner.addColumn(TABLE_NAME, new TableColumn({
       name: 'pickupEvent',
+      isNullable: true,
       type: 'uuid',
     }));
     await queryRunner.createForeignKey(TABLE_NAME, new TableForeignKey({
       columnNames: ['pickupEvent'],
       referencedTableName: 'OrderPickupEvents',
       referencedColumnNames: ['uuid'],
+      onDelete: 'SET NULL',
```

Comment:
The problem I'm getting when I remove this, is that when I try to delete the order pickup event, I get the 'foreign key violates non-null constraints', even though the fields themselves are `{ nullable: true }`.

Reply:
```
 FAIL  tests/merch.test.ts
  ● Test suite failed to run

    QueryFailedError: update or delete on table "OrderPickupEvents" violates foreign key constraint "Orders_pickupEvent_OrderPickupEvents_uuid_fkey" on table "Orders"

      at new QueryFailedError (src/error/QueryFailedError.ts:9:9)
      at PostgresQueryRunner.<anonymous> (src/driver/postgres/PostgresQueryRunner.ts:228:19)
      at step (node_modules/tslib/tslib.js:143:27)
      at Object.throw (node_modules/tslib/tslib.js:124:57)
      at rejected (node_modules/tslib/tslib.js:115:69)
          at runMicrotasks (<anonymous>)
```

Reply:
I think I'll leave this onDelete behavior, but also manually set the orders to be null.

Reply:
I thought this PR would leave the delete pickup event endpoint alone for now and get it working in a followup? What I was saying earlier was that in the delete endpoint we should get all the orders for a pickup event, and update the status to `PICKUP_CANCELED` and the event to the null, and then delete the pickup event, to avoid violating any fkey constraints.

Reply:
Right, we will handle that in a future PR, but when I removed the `onDelete: 'SET NULL'` property, then I was getting the above error. I just realized though, that the error was because of `DatabaseConnection.clear()` deleting all tables asynchronously (link below), so when the `OrderPickupEvents` table was deleted before the `Orders` table, the above error was thrown. I didn't realize the order of the table array mattered, so I'll add a comment on that.

https://github.com/acmucsd/membership-portal/blob/master/tests/data/DatabaseConnection.ts#L28